backends-velox/src/main/scala/io/glutenproject/execution/HashAggregateExecTransformer.scala

```diff
@@ -386,23 +386,33 @@ abstract class HashAggregateExecTransformer(
               val adjustedOrders = veloxOrders.map(sparkOrders.indexOf(_))
               veloxTypes.zipWithIndex.foreach {
                 case (veloxType, idx) =>
-                  val sparkType = sparkTypes(adjustedOrders(idx))
-                  val attr = rewrittenInputAttributes(adjustedOrders(idx))
-                  val aggFuncInputAttrNode = ExpressionConverter
-                    .replaceWithExpressionTransformer(attr, originalInputAttributes)
-                    .doTransform(args)
-                  val expressionNode = if (sparkType != veloxType) {
-                    newInputAttributes +=
-                      attr.copy(dataType = veloxType)(attr.exprId, attr.qualifier)
-                    ExpressionBuilder.makeCast(
-                      ConverterUtils.getTypeNode(veloxType, attr.nullable),
-                      aggFuncInputAttrNode,
-                      SQLConf.get.ansiEnabled)
+                  val adjustedIdx = adjustedOrders(idx)
+                  if (adjustedIdx == -1) {
+                    // The column not found in Spark, and this column is not useful in actual
+                    // calculations, it is sufficient to pass a default DataType value.
+                    val extraAttr = AttributeReference(veloxOrders(idx), veloxType)()
+                    newInputAttributes += extraAttr
+                    val lt = Literal.default(veloxType)
+                    childNodes.add(ExpressionBuilder.makeLiteral(lt.value, lt.dataType, false))
```

Comment:
In Velox, skewness and kurtosis share a common calculation logic. Kurtosis additionally requires the parameter m4, which is always 0 for skewness, rather than null. The Spark skewness buffer does not have the m4 parameter, thus a placeholder m4 with a value of 0 must be passed to Velox, and this value cannot be omitted. Velox will always read m4 column when accessing the intermediate data.

Reply:
Could you add some more comment here to clarify a bit? Thanks!